File: 5_longest_substring/Main.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        String substring  = "";
        int maxLength = 0;
        for (int i = 0; i < s.length(); i++) {
            if (substring.indexOf(s.charAt(i)) == -1) {
                substring += s.charAt(i);
            } else {
                if (substring.length() > maxLength) {
                    maxLength = substring.length();
                }
                int index = substring.indexOf(s.charAt(i));
                substring = substring.substring(index + 1) + s.charAt(i);
            }
        }
        if (substring.length() > maxLength) {
            maxLength = substring.length();
        }
        return maxLength;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        String testString = "abcabcbb";
        int result = solution.lengthOfLongestSubstring(testString);
        System.out.println("Length of the longest substring without repeating characters: " + result);
    }
}

File: 5_longest_substring/main.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        substring = ""
        substring_length = 0

        for char in s:
            if char not in substring:
                substring += char
            else:
                if len(substring) >  substring_length:
                    substring_length = len(substring)
                index_char = substring.find(char)
                substring = substring[index_char + 1:] + char
        if len(substring) > substring_length:
            substring_length = len(substring)
        return substring_length

if __name__ == "__main__":
    s = "dvdf"
    solution = Solution()
    result = solution.lengthOfLongestSubstring(s)
    print(result)